safe-global/safe-wallet-web
# Fix: Hide nonce input if its a spending limit transaction

## What it solves

Resolves #2880 

## How this PR fixes it

- Hides the nonce input field if a non-owner tries to create a spending limit transaction
- Pre-selects the first available token when a non-owner tries to create a spending limit transaction
- Displays the nonce input again if an owner switches from a spending limit token to a non-spending limit token

## How to test it

1. Open a safe with a spending limit as a non-owner
2. Create a transaction and observe the nonce input is hidden
3. Observe a token is pre-selected in the form

---

1. Open a safe as an owner that also has a spending limit
2. Create a transaction
3. Observe the nonce input being visible
5. Select the spending limit token and select spending limit in the radio group
6. Observe the nonce input is hidden
7. Switch to a different token
8. Observe the nonce input is visible

## Screenshots

## Checklist
